Add a command/action to clear history

guywald1 opened the pull request:
This PR adds a command and a button/action to clear the reference history.

Closes https://github.com/microsoft/vscode/issues/73548.

Couldn't find a relevant icon so I made a custom one (top right, second one from the right):

![image](https://user-images.githubusercontent.com/6546430/57563960-496e1b80-73ad-11e9-95b8-6e7fb8a045ae.png)

> I don't know how contributers normally approach icons, so any feedback would be great.

jrieken commented on src/extension.ts:
```diff
@@ -144,6 +144,11 @@ export function activate(context: vscode.ExtensionContext) {
 
+    const clearHistoryCommand = async () => {
+        await clearCommand();
+        history.clear();
```
after clearing the history, the message still reads _"To populate this view, open an editor and run the 'Find All References'-command or run a previous search again:"_ which isn't correct anymore. Instead it should show the default message

jrieken commented on media/action-clear-history.svg:
```diff
@@ -0,0 +1 @@
+<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">	<path fill="#F6F6F6" d="M3.5,12.9c2.6,2.6,6.9,2.6,9.5,0c1.3-1.3,2-3,2-4.8c0-1.8-0.7-3.5-2-4.8C10.6,0.8,6.7,0.7,4,2.9L2.3,1.2		L1.5,2L0.9,8.3l4-0.3l0,0c0.6-0.1,3.9-0.4,3.9-0.4L6.7,5.6C7.9,4.8,9.4,5,10.4,6c0.6,0.6,0.9,1.3,0.9,2.1c0,0.8-0.3,1.5-0.9,2.1		c-1.2,1.2-3.1,1.2-4.2,0L6,9.9L5.9,9.7L5.8,9.6C4.6,8. ...
```
use the same clear icon that's used for clearing results

guywald1 replied:
Wanted to do that, but it seemed like a strange UX that you have an icon to clear results and after clicking it, you have the _same_ icon to clear history. No?

jrieken commented on package.json:
```diff
@@ -136,6 +145,11 @@
+        {
+          "command": "references-view.clearHistory",
+          "group": "navigation",
+          "when": "!reference-list.hasResult"
```
maybe have a context key for showing clear history only when having a history

jrieken commented:
<img width="310" alt="Screenshot 2019-05-13 at 09 55 05" src="https://user-images.githubusercontent.com/1794099/57639637-3736dd80-7565-11e9-9774-cdefe576c82f.png">

Seeing two clear icons now...

guywald1 commented:
@jrieken Yeah, desired behavior but it might be better to only show it when a result isn't selected. Especially if I change the icon to be the same one (WDYT about that btw? Haven't gotten a reply in the thread)

guywald1 commented:
@jrieken For now I changed the icon back and changed it so it doesn't show when there's a result selected.
Though I personally think a different icon is better UX,  but I may be wrong here.

jrieken commented:
> Though I personally think a different icon is better UX, but I may be wrong here.

I understand but we try to go for persistency and use the same icon for the same/similar operation

jrieken commented:
changes look good and work well. thanks!

m-wheeler commented:
Looking forward to it in the next visual studio update!

jrieken commented:
Use Insiders and have it today: https://code.visualstudio.com/insiders/
